pkg/internal/controllers/test/util.go

```diff
@@ -122,48 +122,59 @@ func waitForNotReady(ctx context.Context, cl client.Client, name string) {
 // created namespace as well as a shutdown function to stop the controllers is
 // returned.
 func startControllers(registry *registry.Registry) (context.Context, func(), corev1.Namespace) {
-	ctx, cancel := context.WithCancel(context.Background())
+	log, ctx := ktesting.NewTestContext(GinkgoT())
+	ctx, cancel := context.WithCancel(ctx)
 
 	namespace := corev1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
 			GenerateName: "test-policy-",
 		},
 	}
-	Expect(env.AdminClient.Create(ctx, &namespace)).NotTo(HaveOccurred())
 	By("Created Policy Namespace: " + namespace.Name)
+	Expect(env.AdminClient.Create(ctx, &namespace)).NotTo(HaveOccurred())
+
+	mgrStopped := make(chan struct{})
 
 	shutdown := func() {
+		cancel()
+		<-mgrStopped
+
 		By("Deleting all CertificateRequests after test")
-		env.AdminClient.DeleteAllOf(ctx, new(cmapi.CertificateRequest))
+		Expect(
+			client.IgnoreNotFound(
+				env.AdminClient.DeleteAllOf(context.TODO(), new(cmapi.CertificateRequest)),
+			),
+		).To(Succeed())
```

Comment:
My guess is that the the original authors attempt to delete all CertificateRequests because they can't simply be deleted by deleting their namespace.
We want the controllers in each test to only be reconciling the resources that have been created in that test.